machine is failing while runner is ok

```tf
resource "iterative_machine" "machine" {
    cloud = "azure"
    region = "us-west"
    instance_type = "m"
    #spot = true
    #spot_price = 0.09
} 
```

```sh
Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not string
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: 
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: goroutine 45 [running]:
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: terraform-provider-iterative/iterative/azure.ResourceMachineCreate(0x21ea120, 0xc000508000, 0xc000506000, 0x0, 0x0, 0x1, 0x0)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/iterative/azure/provider.go:28 +0x39bf
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: terraform-provider-iterative/iterative.resourceMachineCreate(0x21ea120, 0xc000508000, 0xc000506000, 0x0, 0x0, 0xc0004bc400, 0x12f8d0a, 0xc00059f240)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/iterative/resource_machine.go:157 +0x6c5
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0004b00c0, 0x21ea0a0, 0xc00020da00, 0xc000506000, 0x0, 0x0, 0x0, 0x0, 0x0)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:285 +0x1ea
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0004b00c0, 0x21ea0a0, 0xc00020da00, 0xc0001c8d20, 0xc00059f240, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:396 +0x67b
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00048c0c0, 0x21ea0a0, 0xc00020da00, 0xc000434550, 0xc00020da00, 0xc000436000, 0x21f9220)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:955 +0x8cf
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative: github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00048cc60, 0x21ea0a0, 0xc00020da00, 0xc0001c8a80, 0xc00048cc60, 0xc000421830, 0xc000075ba0)
2021-01-09T10:11:42.465+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/server/server.go:331 +0xae
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1f8ad60, 0xc00048cc60, 0x21ea160, 0xc000421830, 0xc0004360c0, 0x0, 0x21ea160, 0xc000421830, 0xc0003a6600, 0x1f4)
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003b8c40, 0x21f35a0, 0xc000103200, 0xc000099200, 0xc000404630, 0x29dc740, 0x0, 0x0, 0x0)
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/google.golang.org/grpc/server.go:1194 +0x522
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative: google.golang.org/grpc.(*Server).handleStream(0xc0003b8c40, 0x21f35a0, 0xc000103200, 0xc000099200, 0x0)
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/google.golang.org/grpc/server.go:1517 +0xd05
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00036e2d0, 0xc0003b8c40, 0x21f35a0, 0xc000103200, 0xc000099200)
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/google.golang.org/grpc/server.go:859 +0xa5
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-01-09T10:11:42.466+0100 [DEBUG] plugin.terraform-provider-iterative:       /Users/davidgortega/Documents/projects/@iterative/terraform-provider-iterative/vendor/google.golang.org/grpc/server.go:857 +0x1fd
2021/01/09 10:11:42 [DEBUG] iterative_machine.machine: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalMaybeTainted
2021/01/09 10:11:42 [TRACE] EvalMaybeTainted: iterative_machine.machine encountered an error during creation, so it is now marked as tainted
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalWriteState
2021/01/09 10:11:42 [TRACE] EvalWriteState: removing state object for iterative_machine.machine
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalApplyProvisioners
2021/01/09 10:11:42 [TRACE] EvalApplyProvisioners: iterative_machine.machine has no state, so skipping provisioners
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalMaybeTainted
2021-01-09T10:11:42.468+0100 [WARN]  plugin.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = transport is closing"
2021/01/09 10:11:42 [TRACE] EvalMaybeTainted: iterative_machine.machine encountered an error during creation, so it is now marked as tainted
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalWriteState
2021/01/09 10:11:42 [TRACE] EvalWriteState: removing state object for iterative_machine.machine
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalIf
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalIf
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalWriteDiff
2021/01/09 10:11:42 [TRACE] eval: *terraform.EvalApplyPost
2021/01/09 10:11:42 [ERROR] eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2021/01/09 10:11:42 [ERROR] eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2021/01/09 10:11:42 [TRACE] [walkApply] Exiting eval tree: iterative_machine.machine
2021/01/09 10:11:42 [TRACE] vertex "iterative_machine.machine": visit complete
2021/01/09 10:11:42 [TRACE] dag/walk: upstream of "provider[\"github.com/iterative/iterative\"] (close)" errored, so skipping
2021/01/09 10:11:42 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2021/01/09 10:11:42 [TRACE] dag/walk: upstream of "root" errored, so skipping
2021/01/09 10:11:42 [TRACE] statemgr.Filesystem: not making a backup, because the new snapshot is identical to the old
2021-01-09T10:11:42.468+0100 [DEBUG] plugin: plugin process exited: path=.terraform/plugins/github.com/iterative/iterative/0.6.0/darwin_amd64/terraform-provider-iterative pid=87859 error="exit status 2"
2021/01/09 10:11:42 [TRACE] statemgr.Filesystem: no state changes since last snapshot
2021/01/09 10:11:42 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate
2021/01/09 10:11:42 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2021/01/09 10:11:42 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2021-01-09T10:11:42.480+0100 [DEBUG] plugin: plugin exited



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/examples/main.tf
+++ b/examples/main.tf
@@ -1,0 +1,50 @@
+terraform {
+  required_providers {
+    iterative = {
+      versions = ["0.6"]
+      source = "github.com/iterative/iterative"
+    }
+
+    # iterative = {
+    #   source = "iterative/iterative"
+    #   version = "0.5.7"
+    # }
+  }
+}
+
+provider "iterative" {}
+
+
+resource "iterative_machine" "machine" {
+    cloud = "aws"
+    region = "us-west"
+    instance_type = "t2.micro"
+    spot = true
+    #spot_price = 0.09
+} 
+
+resource "iterative_cml_runner" "runner-gh-aws" {
+    token = "c38c69db3cdb81d2878fcc9c0556ed972249f930"
+    repo = "https://github.com/DavidGOrtega/cml"
+    driver = "github"
+    labels = "tf"
+
+    idle_timeout = 0
+    cloud = "aws"
+    region = "us-west"
+    instance_type = "t2.micro"
+    spot = true
+    #spot_price = 0.09
+} 
+
+# resource "iterative_cml_runner" "runner-gh-aws" {
+#     name = "githubrunner2"
+#     token = "a0b56d03294f0908f4ee7290be2ea53051d227d3"
+#     repo = "https://github.com/DavidGOrtega/3_tensorboard"
+#     driver = "github"
+#     labels = "tf"
+
+#     cloud = "aws"
+#     region = "us-east-2"
+#     instance_type = "t2.micro"
+# } 

--- a/iterative/resource_machine.go
+++ b/iterative/resource_machine.go
@@ -74,6 +74,10 @@ func machineSchema() *map[string]*schema.Schema {
 			Type:     schema.TypeString,
 			Computed: true,
 		},
+		"ssh_public": &schema.Schema{
+			Type:     schema.TypeString,
+			Computed: true,
+		},
 		"ssh_private": &schema.Schema{
 			Type:     schema.TypeString,
 			ForceNew: true,

